Reorganize whole contribution section in a clearier way (#154)
osrd-project/osrd-website: content/docs/guides/contribute/review-process.en.md

Here's a suggested workflow. The quotation marks indicate the **necessity** of communication to ensure the smooth flow of PR validation:

```mermaid
sequenceDiagram
  actor A as PR author
  actor R as Reviewer/Maintainer
  

>   A->>R: ASKS "I need a review"

Reviewer: ```suggestion
  A->>R: ASKS for a review
```

Author: I prefer explicitly write down "I need a review" as "It's approved". Those are things that MUST be clearly said/written IMO.

Reviewer: Both ways are clear and explicit. I find quoting a fictional conversation odd in this context: most technical sequence diagrams [don't use fictional conversations to describe protocols](https://duckduckgo.com/?t=ffab&q=oidc+sequence+diagram&iax=images&ia=images), but instead:

- either describe action performed at each step: `request a review`
- or describe the message itself: `review request`

Author: Nevertheless, I prefer to keep it explicit, in fictional conversation, not actually based on what is done in general, but on the working experience we have within our team where the interpretation of what a "review request" should be in reality. No interpretation here, just a clear & specific message.

>   R->>A: ANSWERS ✅ "Ok" or ❌ "not ok"

Reviewer: ```suggestion
  R->>A: ANSWERS ✅ Approves or ❌ Requests changes
```

Author: I suggest `R->>A: ANSWERS ✅ "Ok, I'll do the review" or ❌ "not ok, I can't do it"`

Reviewer: Is it an interaction even worth mentioning? someone will review it at some point

Author: Same response as above.

>   R-->>+A: COMMENTS, asks for changments

Reviewer: ```suggestion
```

Duplicate of "approves or requests changes

Author: In reality, lines 8 and 9 correspond to the setting up of the agreement between the PR reviewer and producer.

>   R-->>+A: COMMENTS, asks for changments

Author: done

>   A-->>R: ANSWERS to comments or requests changes

Author: done!

>   A-->>-R: DOES necessary changes in dedicated "fixups"

Reviewer: ```suggestion
  A-->>-R: MAKES necessary changes in dedicated "fixups"
```

Author: Yes!

>   R-->>A: REVIEWS & CLOSES request changes and conversations

Reviewer: ```suggestion
  R-->>A: REVIEWS & RESOLVES requested changes and conversations
```

Author: Done!

>   A->>R: DOES rebase & fixup

Reviewer: ```suggestion
  A->>R: REBASE and apply fixups
```

Author: Done!

  R->>A: CHECKS commits

>   R->>A: SAYS ✅ "it's approved"

Reviewer: The process may not complete in a single review round, especially with first time contributions: there has to be a review loop
